frontend-service/src/test/java/com/amazonaws/blox/frontend/integration/StartDeploymentIntegrationTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.frontend.integration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import com.amazonaws.blox.dataservicemodel.v1.exception.ResourceNotFoundException;
+import com.amazonaws.blox.dataservicemodel.v1.model.wrappers.StartDeploymentResponse;
+import com.amazonaws.blox.frontend.operations.StartDeployment;
+import com.amazonaws.serverless.proxy.internal.model.AwsProxyRequest;
+import com.amazonaws.serverless.proxy.internal.model.AwsProxyResponse;
+import com.amazonaws.serverless.proxy.internal.testutils.AwsProxyRequestBuilder;
+import org.junit.Test;
+
+public class StartDeploymentIntegrationTest extends IntegrationTestBase {
```

Comment:
Looking at the amount of coupling we have to do to our web framework's implementation details here, I don't know if this level of testing for each operation is worthwhile.

It's definitely useful to have a fast feedback test that makes sure our spring config is sane, and that all the web framework stuff is wired up correctly in terms of error handling, etc. But if we have to use raw URLs and manual marshaling/unmarshaling for *every* feature we're testing, I'm not sure if the added coupling is worth it.

Do you think it might make sense to keep one fairly generic test here that exercises our routing/error mapping through the spring MVC stuff, and then use the existing testing approach in `operations.*Test` to test actual functionality?

What I've done in other MVC frameworks, is to extract most of the functionality into a controller base class, and then to create a TestController subclass that we can control the behaviour of (e.g. a URL that always throws exceptions, a URL that always redirects, etc) to test these cross-controller concerns. 

Thoughts? Also happy to discuss this in person, if that's more helpful.

Reply:
My initial idea is to use Spring MVC integration test framework, see (https://docs.spring.io/spring-security/site/docs/current/reference/html/test-mockmvc.html)

We could use Spring MVC integration test to test our routing and error handling. But I don't think it will exercise the whole component enough and any effort we save.  The FE routing is not complex as a website, we mostly handle the request just like a Coral API. There is no redirects etc.  The integration test is actual one-one mapping to test each API call. 

If the effort to test the routing is the same, why we don't exercise the whole component from the entry point, lambda handler?  It will not only exercise the Spring mvc routing and also make sure the wiring between lambda to Spring works fine. 

The "raw URLs and manual marshaling/unmarshaling for every feature" is somehow annoying. But I see it's actual the input and output of this component, lambda input and output.   The URL are actually good to use here, as it's how we expose the API. Even we use Spring MVC test, we need to specify the same URL as well.

Reply:
So I went down the MockMVC rabbit hole, and I couldn't get it to work with the aws-serverless-java-container code, because it doesn't actually stand up a full servlet stack from just our Application Config (it [manually wires some spring stuff in the handler](https://github.com/awslabs/aws-serverless-java-container/blob/master/aws-serverless-java-container-spring/src/main/java/com/amazonaws/serverless/proxy/spring/SpringLambdaContainerHandler.java#L52-L63)). But if we could get that working, it does sound like it would be the best of both worlds.

Reply:
You mean you want to start from our` LambdaHandler` class?   The integration test actually write the Lambda container handler as here : https://github.com/blox/blox/blob/eb90b05d93f250dca8de3c2074c47cb52ff514f3/frontend-service/src/test/java/com/amazonaws/blox/frontend/integration/IntegrationTestConfig.java#L46-L47

We cannot provide a configuration Class (a test one) to start because LambadaContainerHandler will initialize the Spring app context but not provide a way to get it out, so we cannot wire our test into the context.